app/controllers/api/v1/menu_controller.rb

```diff
@@ -0,0 +1,57 @@
+class API::V1::MenuController < ApplicationController
+
+  def index
+    # TODO: Unwrap child menu items into resulting json data
+    render :json => MenuItem.where(menu_item_id: nil), :include => [:menu_items]
+
+  end
+
+  def create
+    restrict_access
+
+    menu_item = MenuItem.new(item_params)
+    menu_item.save
+
+    redirect_to api_v1_menu_url(menu_item)
+  end
+
+  def show
+    render :json => MenuItem.find(params[:id])
+  end
+
+  def update
+    restrict_access
+
+    menu_item = MenuItem.find(params[:id])
+
+    if menu_item.update(item_params)
+      redirect_to api_v1_menu_url(menu_item)
+    else
+      raise 'Unable to save menu item'
+    end
+  end
+
+  def destroy
+    restrict_access
+
+    menu_item = MenuItem.find(params[:id])
+    menu_item.destroy
+
+    # What is preferably returned here?
+    render :json => []
+  end
+
+  private
+
+  def item_params
+    params.require(:item).permit(:title, :active)
```

Comment:
This does not allow anyone to create nested menus using the REST API. This could be included in another release if you wish since it requires more extensive tests before including the functionality. **Please write a comment reflecting the limitation if you choose to not include it.**